### PR TITLE
Add CargoCondition(s) to Passenger

### PR DESCRIPTION
**Pull request by MustaphaTR**
This was wanted to make infantry in pillbox not gain veterancy for a while, i coded that to make Internet Center work on Generals Alpha, wanted to get to upstream too, as this can be pretty useful for modders.

### CODE REVIEW

**Comment by GraionDilach** [comment]
Can't you shift this to an interface instead of ticking constantly?

**Comment by pchote** [comment]
Yes, please do that.  You can call the interface from `Cargo` when the passenger is added/removed.

**Comment by MustaphaTR** [comment]
Updated.

**Review by abcdefg30 — COMMENTED** [review]
Seems like you forgot to push the changes made to call the new interfaces.

**Comment by MustaphaTR** [comment]
I forgot to say "updated" here again, so writing in case this got missed.

**Comment by pchote** [comment]
Ping @Smittytron: Do we want to use this to disable or reduce experience for units in RA pillboxes?

**Review by GraionDilach — APPROVED** [review]
:+1:

File: OpenRA.Mods.Common/Traits/Passenger.cs [hunk 1]
```diff
@@ -94,6 +115,28 @@ public string VoicePhraseForOrder(Actor self, Order order)
 			return Info.Voice;
 		}
 
+		void INotifyEnteredCargo.OnEnteredCargo(Actor self, Actor cargo)
+		{
+			string specificCargoCondition;
+			if (conditionManager != null)
+			{
+				if (anyCargoToken == ConditionManager.InvalidConditionToken && Info.CargoCondition != null)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Passenger.cs]: This should use `string.IsNullOrEmpty()` instead of checking null directly.

**Review by pchote — DISMISSED** [review]
:+1: after one minor fix:

**Comment by MustaphaTR** [comment]
Updated.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/4fb7f35f9027048f46363860b322f2f1d72dd28f

**Pull request merged**